tests/apollo/util/skvbc.py

```diff
@@ -221,6 +221,10 @@ async def assert_kv_write_executed(self, key, val):
             assert {key: val} == kv_reply, \
                 f'Could not read original key-value in the case of n={config.n}, f={config.f}, c={config.c}.'
 
+    async def wait_for_liveness(self):
+        key, value = await self.write_known_kv()
+        await self.assert_kv_write_executed(key, value)
```

Comment:
To make this deterministic, you'll need to add retries. For example, if the system is not yet ready to process requests, the `await self.write_known_kv()` is going to fail.
See an example of how we can add a retrying `while True` loop here:
https://github.com/vmware/concord-bft/blob/master/tests/apollo/util/bft.py#L972

Reply:
Thanks will add this!